skbeam/core/accumulators/projector.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env python
+
+from __future__ import division
+import numpy as np
+
+"""
+authors: Mikhail Dubrovin, TJ Lane, Christopher O'Grady
+"""
+
+
+class Projector(object):
+
+    def __init__(self, bin_values, nbins, weights=None, norm=True):
+        """
+        Parameters
+        ----------
+        bin_values : np.ndarray
+            For each pixel, this is the value of that determines which bin
+            a pixel's intensity will contribute to
+        weights : np.ndarray
+            A weight for each pixel with the same shape as bin_values (set
+            to zero to ignore a pixel)
+        nbins : int
+            The number of bins to project into
+        """
+
+        self.bin_values = bin_values
+        self.weights = weights
+        self.nbins = nbins
+        self.norm = norm
+
+        # _ibv = included bin values
+        # this is used to select only unmasked pixels in the analysis
+        # (i.e. those with weight!=0)
+        if weights is not None:
+            self._ibv = (weights != 0)
+        else:
+            self._ibv = np.s_[:]  # slice that returns arr unchanged
```

Comment:
I have restructured the code to eliminate the np.s_[:] and reduce the number of flatten() calls (all except one are in the calls to numpy.bincount)